### PR TITLE
[RS] Add Face API examples into spec.

### PR DESCRIPTION
**Pull request by leonhsl**
BUG=XWALK-6403

### CODE REVIEW

**Comment by leonhsl** [comment]
@huningxin @halton PTAL, Thanks~

File: spec/face.html [hunk 1]
```diff
@@ -1098,6 +1098,122 @@
...
+      </h2>
+      <h3>
+        Start/Stop face module
+      </h3>
+      <pre class='example highlight'>
+        var previewStream;
+        var ft;
+
+        // Start stream firstly, then start face module.
+        startButton.onclick = function(e) {
```
Comment (huningxin) [inline spec/face.html:1119]: It would be better to define `startButton`, `stopButton` in the begin of the file.

File: spec/face.html [hunk 2]
```diff
@@ -1098,6 +1098,122 @@
...
+                };
+        
+                ft.onended = function(e) {
+                  console.log('Face module ends without stop');
+                };
+        
+                ft.ready = false;
+                ftStarted = false;
+                onGetConfButton();
+              }, errorCallback);
```
Comment (huningxin) [inline spec/face.html]: Please define a very simple `errorCallback`.

Comment (huningxin) [inline spec/face.html]: Please define a basic `errorCallback`.

File: spec/face.html [hunk 3]
```diff
@@ -1098,6 +1098,122 @@
...
+                  ft.start().then(
+                      function() {
+                        console.log('Face module start succeeds');},
+                      function(e) {
+                        console.log('Face module start failed: ' + e.message);}); 
+                };
+        
+                ft.onprocessedsample = function(e) {
+                  console.log('Got face module processedsample event.');
+                  ft.getProcessedSample(false, false).then(function(processed_sample) {
```
Comment (huningxin) [inline spec/face.html]: According to our coding style, please use `processedSample`.

File: spec/face.html [hunk 4]
```diff
@@ -1098,6 +1098,122 @@
...
+                    console.log('Got face module processedsample data.');
+                    // Use processed_sample.faces data to work for you.
```
Comment (huningxin) [inline spec/face.html]: Please give a very basic sample of `processedSample.faces` usage, say print how many faces are detected.

File: spec/face.html [hunk 5]
```diff
@@ -1098,6 +1098,122 @@
...
+              function(e) {
+                console.log('Face module stop failed');
+                stopPreviewStream();});
+        }
+      </pre>
+      <h3>
+        Face module configuration.
+      </h3>
+      <pre class='example highlight'>
+        // Set configuration. Please prepare confData value firstly.
```
Comment (huningxin) [inline spec/face.html]: Please provide a specific example of configuration, say enable/disable landmarks tracking.

**Comment by leonhsl** [comment]
Thanks for kindly review! Addressed all comments and updated, PTAL, Thanks~

File: spec/face.html (hunk 2 above)
Comment (huningxin) [inline spec/face.html]: @leonhsl , how about this one?

Comment (huningxin) [inline spec/face.html]: This one is fixed. Please ignore my comment.

File: spec/face.html [hunk 6]
```diff
@@ -1098,6 +1098,143 @@
...
+          // Disable landmarks.
+          landmarks: {
+            enable: false
+          }, 
+          // Enable recognition.
+          recognition: {
+            enable: true
+          }
+        };
+        ft.configuration.set(confData).then(
```
Comment (huningxin) [inline spec/face.html]: I don't quite get the use case that calling `set`, `get` and `getDefaults` in sequence. Could you please put them into corresponding button onclick handler? I think they are close to real use case. WDYT?

Comment (leonhsl) [inline spec/face.html]: Yeah, Ack, Will modify soon.

**Comment by leonhsl** [comment]
Addressed comments and updated, PTAL again, Thanks~

File: spec/face.html [hunk 7]
```diff
@@ -1098,6 +1098,153 @@
...
+                };
+        
+                ft.onprocessedsample = function(e) {
+                  console.log('Got face module processedsample event.');
+                  ft.getProcessedSample(false, false).then(function(processedSample) {
+                    console.log('Got face module processedsample data.');
+                    // Use processedSample.faces data to work for you.
+                    console.log('Detected faces number: ' + processedSample.faces.length);
+                    // You can get all avaiable detection/landmarks/recognition data
+                    // of every face from processedSample.faces.
```
Comment (huningxin) [inline spec/face.html:1152]: How about adding "Please refer to FaceData interface."

File: spec/face.html [hunk 8]
```diff
@@ -1098,6 +1098,153 @@
...
+      <h3>
+        Face module configuration.
+      </h3>
+      <pre class='example highlight'>
+        var setConfButton = document.getElementById('setConfButton');
+        var getConfButton = document.getElementById('getConfButton');
+        var getDefaultConfButton = document.getElementById('getDefaultConfButton');
+
+        // Set configuration. Simple configuration example as bellow.
+        var confData = {
```
Comment (huningxin) [inline spec/face.html:1209]: How about adding comment like "Please refer to FaceConfigurationData interface for confData details."

**Comment by huningxin** [comment]
Thanks for updating! I am thinking about adding reference to interface would help developers to get details. Please take a look at my comments. Thanks!

**Comment by leonhsl** [comment]
Added reference to interface to address comments, PTAL again, Thanks~

**Comment by huningxin** [comment]
LGTM. Merging.

**Pull request merged**